packages/docker/src/executors/release-publish/release-publish.impl.ts

```diff
@@ -0,0 +1,93 @@
+import { ExecutorContext, ProjectGraphProjectNode, logger } from '@nx/devkit';
+import { execSync } from 'child_process';
+import {
+  DockerReleasePublishSchema,
+  NormalizedDockerReleasePublishSchema,
+} from './schema';
+import { join } from 'path';
+import { existsSync, readFileSync } from 'fs';
+
+export default async function dockerReleasePublish(
+  schema: DockerReleasePublishSchema,
+  context: ExecutorContext
+) {
+  logger.warn(
+    `Docker support is experimental. Breaking changes may occur and not adhere to semver versioning.`
+  );
+  const projectConfig = context.projectGraph.nodes[context.projectName];
+  const options = normalizeOptions(projectConfig, schema);
+  if (!options.dryRun) {
+    const digest = dockerPush(options.imageReference);
+    logger.log(
+      `Successfully pushed ${options.imageReference}. Digest: ${digest}`
+    );
+  } else {
+    logger.log(
+      `Docker Image ${options.imageReference} was not pushed as --dry-run is enabled.`
+    );
+  }
+  return {
+    success: true,
+  };
+}
+
+function normalizeOptions(
+  projectConfig: ProjectGraphProjectNode,
+  schema: DockerReleasePublishSchema
+): NormalizedDockerReleasePublishSchema {
+  return {
+    imageReference: findImageReference(projectConfig, schema),
+    dryRun: process.env.NX_DRY_RUN === 'true' || schema.dryRun || false,
+  };
+}
+
+function findImageReference(
+  projectConfig: ProjectGraphProjectNode,
+  schema: DockerReleasePublishSchema
+) {
+  let imageRef = readVersionFromFile(projectConfig.data.root);
+  if (imageRef) {
+    if (checkDockerImageExistsLocally(imageRef)) {
+      return imageRef;
+    }
+    throw new Error(
+      `Could not find Docker Image ${imageRef}. Did you run 'nx release version'?`
+    );
+  }
+}
+
+function readVersionFromFile(projectRoot: string) {
+  const versionFilePath = join(projectRoot, '.docker-version');
+  if (!existsSync(versionFilePath)) {
+    throw new Error(
+      "Could not find .docker-version file. Did you run 'nx release version'?"
+    );
+  }
+
+  const version = readFileSync(versionFilePath, { encoding: 'utf8' });
+  return version.trim();
+}
+
+function checkDockerImageExistsLocally(imageRef: string) {
+  try {
+    const result = execSync(
+      `docker images --filter "reference=${imageRef}" --quiet`,
+      { encoding: 'utf8' }
+    );
+    return result.trim().length > 0;
+  } catch {
+    return false;
+  }
+}
+
+function dockerPush(imageReference: string) {
+  try {
+    const result = execSync(`docker push ${imageReference} --quiet`, {
```

Comment:
Remove --quiet so that we're more transparent to what docker users would be used to.

If they pass --quiet, we should pass it through. aka Add an option and pass it through here. Users can configure it easily with targetDefaults